common/ledger/blkstorage/snapshot_test.go

```diff
@@ -0,0 +1,465 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package blkstorage
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/common/ledger/snapshot"
+	"github.com/hyperledger/fabric/common/ledger/testutil"
+	"github.com/hyperledger/fabric/internal/pkg/txflags"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testBlockDetails struct {
+	txIDs           []string
+	validationCodes []peer.TxValidationCode
+}
+
+func TestBootstrapFromSnapshot(t *testing.T) {
```

Comment:
While the test function name is `BootStrapFromSnapshot()`, it does the following

1. Exports TxID from original store.
2. Builds a new store from the exported TxIDs
3. Checks for all query including getBlockByNum, hash, etc...
4. Adds a few new blocks and check query again.
5. Again exports TxIDs from the newly bootstrapped store to see whether it can export.
6. Checks whether the index is being synced during recovery. 

It certainly took me sometime to review this. Based on my experience, in the future, when we update the related code, it would take sometime to understand this test and make necessary changes.  If we are able to split this into some sub-test/sub-function, I believe that it would have improved the readability significantly. Is there any challenge in splitting this test and organizing it better? I am fine if it can be done in a subsequent PR. 

(1, 2, 3, 4) - group 1
(5) - group 2
(6) - group 3

Reply:
Yes, agreed. I'll break this up in next PR.